### ISSUE TITLE
`deprecated` decorator doesn't work on implementation when function has overloads

### ISSUE THREAD

**Issue by DetachHead** [microsoft/pyright#5892]
## Environment data

-   Language Server version: 2023.8.50
-   OS and version: windows 10
-   Python version (& distribution if applicable, e.g. Anaconda): 3.11

## Code Snippet

```python
from typing import overload

from typing_extensions import deprecated


@overload
def foo(value: int):
    ...


@overload
def foo(value: str):
    ...


@deprecated("asdf")
def foo(value: object):
    ...


foo(1)
foo("")
```

## Expected behavior

decorating the implementation with `@deprecated` should warn on every usage of the function, and decorating an individual overload should only warn on usages of that overload.

otherwise if you want to mark the entire function as deprecated, you have to duplicate the deprecation message across every overload

## Actual behavior

no deprecation warnings on either usages of the function

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.326, which I just published. It will also be included in a future release of pylance.
